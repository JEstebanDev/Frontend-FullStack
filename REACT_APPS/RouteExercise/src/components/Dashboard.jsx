import React from "react";
import { Link, Outlet, useNavigate } from "react-router-dom";

export default function Dashboard() {
  const navigate = useNavigate();
  return (
    <div>
      <div className="container">
        <h1>FullStackBootCamp</h1>
      </div>
      <button
        onClick={() => {
          navigate(-1);
        }}
      >
        Go back
      </button>
      <Link to={"userPage"}>Go to user</Link>
      <Outlet />
    </div>
  );
}
