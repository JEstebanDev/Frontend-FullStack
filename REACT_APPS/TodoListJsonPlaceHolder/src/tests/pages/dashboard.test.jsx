import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Dashboard from "../../pages/dashboard";

describe("testing the component", () => {
  test("should render the Dashboard and take snapshot", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Dashboard />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
