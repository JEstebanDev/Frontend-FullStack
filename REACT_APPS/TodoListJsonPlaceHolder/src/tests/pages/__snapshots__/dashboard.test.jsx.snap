// Vitest Snapshot v1

exports[`testing the component > should render the Dashboard and take snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="navbar animate__animated animate__bounceInDown"
    >
      <div
        class="logo"
      >
        <img
          src="/src/assets/vite.svg"
          width="50px"
        />
        <h3>
          TODO List
        </h3>
      </div>
      <div
        class="nav-links"
      >
        <a
          href="/list"
        >
          Dashboard 
          <i
            class="ri-list-check"
          />
        </a>
        <a
          href="/create"
        >
          Create a todo 
          <i
            class="ri-add-circle-fill"
          />
        </a>
      </div>
    </div>
    <div
      class="content"
    />
  </div>
</div>
`;
